src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -4770,9 +4770,16 @@ void CodeGen::genCheckUseBlockInit()
     CLANG_FORMAT_COMMENT_ANCHOR;
 
 #ifdef TARGET_64BIT
+#if defined(TARGET_AMD64)
 
-    genUseBlockInit = (genInitStkLclCnt > (largeGcStructs + 8));
+    // We can clear using aligned SIMD so the threshold is lower,
+    // and clears in order which is better for auto-prefetching
+    genUseBlockInit = genInitStkLclCnt > (largeGcStructs + 4);
 
+#else // !defined(TARGET_AMD64)
+
+    genUseBlockInit = (genInitStkLclCnt > (largeGcStructs + 8));
+#endif
```

Comment:
This change requires a corresponding change in Compiler::fgVarNeedsExplicitZeroInit . That should result in additional code size improvements.

Reply:
Changed